### PR TITLE
Improve completions after configurable keyword in ModulePart cxt

### LINKED ISSUE #36323: Type suggestions are not provided after `configurable` keyword when its followed by a function
Consider the following scenario. 
```
configurable <cursor>

public function main() {
    
}
``` 
Here the completion items include module level completions items (like main function, type defs, service declarations, etc). This needs to be fixed to suggest types instead.

_Originally posted by @mindula in https://github.com/ballerina-platform/ballerina-lang/issues/36189#issuecomment-1131190584_

### PR DESCRIPTION
**Pull request by LakshanWeerasinghe**
## Purpose
$subject

Fixes #36323

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/completion/ModulePartNodeContextTest.java [hunk 1]
```diff
@@ -34,6 +38,12 @@ public void init() throws Exception {
         preLoadAndInit();
     }
 
+    @Test(dataProvider = "completion-data-provider")
+    @Override
+    public void test(String config, String configPath) throws WorkspaceDocumentException, IOException {
+        super.test(config, configPath);
+    }
+
```
Comment (malinthar) [inline language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/completion/ModulePartNodeContextTest.java:46]: Let's remove this unwanted changes

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ModulePartNodeContext.java [hunk 2]
```diff
@@ -56,7 +56,6 @@ public ModulePartNodeContext() {
     @Override
     public List<LSCompletionItem> getCompletions(BallerinaCompletionContext context, ModulePartNode node) {
         List<LSCompletionItem> completionItems = new ArrayList<>();
-
```
Comment (malinthar) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/ModulePartNodeContext.java:59]: Let's remove this unwanted formatting changes

**Review by malinthar — CHANGES_REQUESTED** [review]
1. We have to figure out the `type descriptors` allowed to be configurable.
Thereafter, we have to decide if we are going to sort them to the top or provide completions only for allowed `type descriptors` + modules.

2. Shall we add test cases covering following cases as well.

case 1

``` ballerina
import ballerina/module1;

configurable s<cursor>

function getInt() returns int {
    return 12;
}
```
case 2

```ballerina
import ballerina/module1;

configurable module1:<cursor>

function getInt() returns int {
    return 12;
}
```
case 3

``` ballerina
import ballerina/module1;

configurable <cursor>
int a = 10;

function getInt() returns int {
    return 12;
}
```

File: language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/completion/ModulePartNodeContextTest.java (hunk 1 above)
Comment (LakshanWeerasinghe) [reply language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/completion/ModulePartNodeContextTest.java:46]: @malinthar If we remove this the test case won't execute during the build. Am I correct?

Comment (malinthar) [reply language-server/modules/langserver-core/src/test/java/org/ballerinalang/langserver/completion/ModulePartNodeContextTest.java:46]: We already have a data provider with the same name in the class. So we don't need this ryt?

**Pull request closed without merging**